src/main/java/org/opensrp/service/ImportBulkDataService.java

```diff
@@ -0,0 +1,292 @@
+					practitioner.setUsername(userName);
+
+					practitionerRole = convertToPractitionerRole(practitionerIdentifier, csvdata);
+					if (organizationIdentifier != null) {
+						code = practitionerRole.getCode() != null ? practitionerRole.getCode().getText() : null;
+						practitionerRole.setOrganizationIdentifier(organizationIdentifier);
+						practitionerService.addOrUpdatePractitioner(practitioner);
+						practitionerRoleService
+								.assignPractitionerRole(organizationId, practitionerIdentifier, code, practitionerRole);
+						rowsProcessed++;
+					}
+				} else if (organizationIdentifier == null) {
+					failedRecordSummaries.add(getFailedRecordSummaryObject(rowCount,"Failed to get organization against the given organization Id"));
+				} else {
+					failedRecordSummaries.add(getFailedRecordSummaryObject(rowCount,"Validation failed, provided organization name mismatches with the system"));
+				}
+			}
+			catch (Exception e) {
+				logger.error(
+						"Exception occurred while persisting practitioner and pratitioner role data : " + e.getMessage());
+				failedRecordSummaries.add(getFailedRecordSummaryObject(rowCount, "Exception occurred due to : " + e.getMessage()));
+			}
+		}
+		csvBulkImportDataSummary.setNumberOfRowsProcessed(rowsProcessed);
+		csvBulkImportDataSummary.setFailedRecordSummaryList(failedRecordSummaries);
+		return csvBulkImportDataSummary;
+	}
+
+	private Boolean validateLocationData(Map<String, String> csvdata) {
+		String locationIdFromCSV;
+		String locationNameFromCSV;
+		PhysicalLocation physicalLocation;
+
+		locationNameFromCSV = getValueFromMap(LOCATION_NAME_KEY, csvdata);
+
+		if (csvdata.containsKey(LOCATION_ID_KEY)) {
+			locationIdFromCSV = getValueFromMap(LOCATION_ID_KEY, csvdata);
+			physicalLocation = physicalLocationService.getLocation(locationIdFromCSV, false);
+			if (locationNameFromCSV != null && physicalLocation != null && physicalLocation.getProperties() != null
+					&& physicalLocation.getProperties()
+					.getName().equals(locationNameFromCSV)) {
+				return Boolean.TRUE;
+			}
+		}
+		return Boolean.FALSE;
+	}
+
+	private Boolean validateOrganizationFields(Map<String, String> csvdata) {
+		String organizationIdFromCsv;
+		Long organizationId = 0l;
+		String organizationNameFromCSV = "";
+		Organization organization;
+
+		organizationNameFromCSV = getValueFromMap(ORGANIZATION_NAME_KEY, csvdata);
+
+		if (csvdata.containsKey(ORGANIZATION_ID_KEY)) {
+			organizationIdFromCsv = csvdata.get(ORGANIZATION_ID_KEY);
+			organizationId = Long.valueOf(organizationIdFromCsv);
+			organization = organizationService.getOrganization(organizationId);
+			if (organization != null && organization.getName().equals(organizationNameFromCSV)) {
+				return Boolean.TRUE;
+			}
+		}
+		return Boolean.FALSE;
+	}
+
+	private String getValueFromMap(String key, Map<String, String> map) {
+		if (map.containsKey(key)) {
+			return map.get(key);
+		}
+		return null;
+	}
```

Comment:
Not sure this is required since `map.get(key)` will return null if key does not exist